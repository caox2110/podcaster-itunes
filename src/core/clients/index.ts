export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
