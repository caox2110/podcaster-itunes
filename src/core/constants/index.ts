import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
