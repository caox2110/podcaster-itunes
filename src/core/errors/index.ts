export { default as ExceptionHandler } from './Exceptions.handler';
