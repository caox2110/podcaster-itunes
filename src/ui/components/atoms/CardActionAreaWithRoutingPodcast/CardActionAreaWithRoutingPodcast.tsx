import { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';

import { MuiCardActionArea } from '..';

interface CardActionAreaWithRoutingPodcastType {
  /**
   * Children react element
   *
   * @type {React.ReactNode}
   */
  children: React.ReactNode;
  /**
   * Route go to
   *
   * @type {(string | number)}
   */
  route: string;
}

export default function CardActionAreaWithRoutingPodcast({
  children,
  route,
}: CardActionAreaWithRoutingPodcastType): JSX.Element {
  const navigate = useNavigate();
  const handleClickEvent = useCallback(() => {
    navigate(route);
  }, [navigate, route]);
  return <MuiCardActionArea onClick={handleClickEvent}>{children}</MuiCardActionArea>;
}
