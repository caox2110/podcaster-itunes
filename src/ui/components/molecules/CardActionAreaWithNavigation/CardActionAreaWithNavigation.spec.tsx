import { fireEvent, render, screen } from '@/core/test';
import type { RenderResult } from '@/core/test';

import CardActionAreaWithRoutingPodcast, {
  CardActionAreaWithNavigationType,
} from './CardActionAreaWithNavigation';

type RenderAppType = {
  link: HTMLLinkElement;
} & RenderResult;

const properties = {
  children: <h1>Card Action</h1>,
  route: '/podcast/1633466636',
};

function renderApp({ ...rest }: CardActionAreaWithNavigationType = properties): RenderAppType {
  const utils = render(<CardActionAreaWithRoutingPodcast {...rest} />, { withRouter: true });

  const link = screen.getByRole(/link/) as HTMLLinkElement;

  return {
    ...utils,
    link,
  };
}

describe('CardActionAreaWithRoutingPodcast', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });

  describe('Default behaviors', function () {
    it('should go to new route after call onClick event in the child', function () {
      const handleClick = jest.fn();
      const { link } = renderApp({
        ...properties,
        onClick: handleClick,
      });

      expect(screen.getByText(/card action/i)).toBeInTheDocument();

      fireEvent.click(link);
      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });
});
