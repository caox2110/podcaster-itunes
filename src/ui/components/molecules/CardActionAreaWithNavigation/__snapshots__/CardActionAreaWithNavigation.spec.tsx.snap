// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardActionAreaWithRoutingPodcast Rendering Should render without throwing any errors 1`] = `
<div>
  <a
    class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    href="/podcast/1633466636"
    tabindex="0"
  >
    <h1>
      Card Action
    </h1>
    <span
      class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </a>
</div>
`;
